src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -0,0 +1,551 @@
+
+    def _acquireTiles(self):
+        """
+         Acquire needed tiles by moving the stage to the tile position then calling acq.acquire
+        :return: (list of list of DataArrays): list of acquired data for each stream on each tile
+        """
+        da_list = []  # for each position, a list of DataArrays
+        trep = (self._nx, self._ny)
+        prev_idx = [0, 0]
+        i = 0
+        for ix, iy in self._generateScanningIndices(trep):
+            logging.debug("Acquiring tile %dx%d", ix, iy)
+            self._moveToTile((ix, iy), prev_idx, self._sfov)
+            prev_idx = ix, iy
+
+            def acquireTile():
+                """
+                Calls acquire function and blocks until the data is returned.
+                :return (list of DataArrays): list of acquired das for the current tile
+                """
+                # Update the progress bar
+                self._future.set_progress(end=self.estimateTime(self._nb - i) + time.time())
+
+                self._future.running_subf = acq.acquire(self._streams, self._settings_obs)
+                das, e = self._future.running_subf.result()  # blocks until all the acquisitions are finished
+                if e:
+                    logging.warning("Acquisition for tile %dx%d partially failed: %s",
+                                    ix, iy, e)
+
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+                return das
+
+            das = acquireTile()
+
+            if i == 0:
+                # Check the FoV is correct using the data, and if not update
+                self._sfov = self._checkFov(das, self._sfov)
+                # Measure initial optical focus (to be used to compare next tiles focus)
+                if self._fm_focus_stream:
+                    try:
+                        # Use the same image index as focus stream
+                        self._optical_focus = MeasureOpticalFocus(das[self._streams.index(self._fm_focus_stream)])
+                    except IndexError:
+                        logging.warning("Failed to get image to measure focus on.")
+            else:
+                if hasattr(self, '_optical_focus'):
+                    try:
+                        current_focus = MeasureOpticalFocus(das[self._streams.index(self._fm_focus_stream)])
+                        # Run the autofocus if current focus got worse than permitted deviation
+                        if abs(current_focus - self._optical_focus) / self._optical_focus > self._focus_fidelity:
+                            self._future.running_subf = AutoFocus(self._fm_focus_stream.detector,
+                                                                  self._fm_focus_stream.emitter,
+                                                                  self._fm_focus_stream.focuser,
+                                                                  good_focus=self._optical_focus)
```

Comment:
Here is missing the small range as mentioned in the spec, just how small it should be?